src/gui/sdlmain.cpp

```diff
@@ -2957,6 +2963,8 @@ static void setup_initial_window_position_from_conf(const std::string& window_po
 		            window_position_val.c_str(),
 		            desktop.w,
 		            desktop.h);
+
+		set_auto_window_position_setting();
```

Comment:
This is again mixing concerns about what the config system should be.  The first use-case was good.  Parse the value with `sscanf` and if the value is garbage, fall back to a default.

Here though, you're checking against the user's runtime environment (the user's desktop size) which could change at any time if the user changes desktop resolution (some people probably run DOSBox on a CRT and wish to do this) or move DOSBox to a different monitor.

Say, for example, DOSBox opens on the wrong monitor.  Oops, resolution is too small so we fall back to default.  User moves it to the correct monitor and changes some other setting which causes the window to be re-created.  Except now it gets re-created at the small default size because we've thrown away their config setting.

IMO we should only change config settings if they are invalid, not if the users's screen is too small or their graphics driver fails.  If we wanted this value to be the "current window resolution", we would be updating this every time the window resizes which we aren't doing.

Reply:
Sorry, this is window position, not resolution but the same point still applies.

Reply:
Pasting the code in question here to make sure we're talking about the same thing:

```cpp
        const auto desktop = get_desktop_size();

	const bool is_out_of_bounds = x < 0 || x > desktop.w || y < 0 ||
	                              y > desktop.h;
	if (is_out_of_bounds) {
		LOG_WARNING("DISPLAY: Invalid 'window_position' setting: '%s'. "
		            "Requested position is outside the bounds of the %dx%d "
		            "desktop, using 'auto'.",
		            window_position_val.c_str(),
		            desktop.w,
		            desktop.h);

		set_auto_window_position_setting();
		return;
	}
```

I think you're right, and by changing the setting to `auto`, I've changed the behaviour, like you described.

To be fair, It's a pretty change edge case-y change in behaviour, but a change it is, so thanks for pointing it out 😄 Again, I was just going by the log message ("using `auto`"), without thinking about it too much. It's good to have you to keep me on my toes, hey 😎

Although it most cases I'd say syncing the config settings is desirable, I see your point that in some rare edge cases that's worse, this is being one of them.

Btw, it was me who originally added the `window_position` property, that was actually my first Staging PR ever 😄 My motivation was to use it in conjunction with `window_decorations = off` (my doing too) to set the DOSBox window to occupy exactly the left half of the screen when playing RPGs with my mapper program, [Gridmonger](https://gridmonger.johnnovak.net/). I'm actually doubtful people would rely on that edge case you described, but fair point, I'll remove the syncing in this case.

Check this example out:
https://gridmonger.johnnovak.net/_static/img/frontpage/ss-eob-full.png